### PR TITLE
[Enhancement] refactor TaskWorkerPool

### LINKED ISSUE #19575: [Enhancement] refactor TaskWorkerPool
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, doris put a lot tasks in TaskWorkerPool and init an instance for every kind task. We should refactor TaskWorkerPool to be more object oriented.

### Solution

- [ ] TaskWorkerPool base class.
- [ ] XXTaskWorkerPool for every kind, e.g. ReportTaskWorkerPool, PushTaskWorkerPool
- [ ] Init Corresponding WorkerPool for every kind.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by bobhan1**
# Proposed changes

Issue Number: close #19575 

Refactor the TaskWorkerPool from `switch-case` of process-oriented paradigm to object-oriented paradigm

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.45 seconds
 stream load tsv:          424 seconds loaded 74807831229 Bytes, about 168 MB/s
 stream load json:         25 seconds loaded 2358488459 Bytes, about 89 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230512152433_clickbench_pr_143074.html

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run p1

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run buildall

File: be/src/agent/task_worker_pool.cpp [hunk 1]
```diff
@@ -289,6 +192,8 @@ void TaskWorkerPool::notify_thread() {
     VLOG_CRITICAL << "notify task worker pool: " << _name;
 }
 
+void TaskWorkerPool::_set_callback_and_worker_count() {}
```
Comment (dataroaring) [inline be/src/agent/task_worker_pool.cpp:195]: This method should not be called. so we should put a CHECK(false) here.

**Pull request closed without merging**